### ISSUE TITLE
Integrate Superluminal

### ISSUE THREAD

**Issue by JonnyOThan** [KSPModdingLibs/KSPProfiler#3]
This could be useful... https://www.nuget.org/packages/SuperluminalPerf/

Superluminal supports Unity out of the box, but not the version that KSP uses.

**Comment by gotmachine** [comment]
Well, upgrading the unity version KSP uses is definitely quite difficult.
On a side note, lately I've been using DotTrace extensively with great results in simple manual capture mode, but it also has an API that could in theory allow for targeted profiling of specific code.

**Comment by JonnyOThan** [comment]
Right, I looked briefly into updating KSP's unity version and I don't think it's really feasible.  But Superluminal has an instrumentation API we could use (and the thing I linked above is C# bindings for it, though I'm not sure if it's compatible with .net 4.5).

I'll have to check out DotTrace as well.  Perhaps consider this issue more as a general "integrate with external profilers" item.

Superluminal is one of the best tools I've used for native code - it has a sampling profiler with 8khz sample rate where most are 1khz.  The only thing better I've seen is intel's vtune but it can only be used on intel chips.
